cli/src/command/account.rs

```diff
@@ -920,81 +956,109 @@ pub async fn voting_output_command(account: &Account) -> Result<(), Error> {
-        .iter()
-        .find(|a| a.key_index() == address.key_index() && a.internal() == address.internal())
-    {
-        output_ids = address.output_ids().as_slice();
-
-        for output_id in output_ids {
-            if let Some(output_data) = account.get_output(output_id).await {
-                // Output might be associated with the address, but can't be unlocked by it, so we check that here.
-                // Panic: cannot fail for outputs belonging to an account.
-                let (required_address, _) = output_data
-                    .output
-                    .required_and_unlocked_address(current_time, output_id, None)
-                    .unwrap();
-
-                if address.address().as_ref() == &required_address {
-                    if let Some(nts) = output_data.output.native_tokens() {
-                        native_tokens.add_native_tokens(nts.clone())?;
-                    }
-                    match &output_data.output {
-                        Output::Nft(nft) => nfts.push(nft.nft_id_non_null(output_id)),
-                        Output::Alias(alias) => aliases.push(alias.alias_id_non_null(output_id)),
-                        Output::Foundry(foundry) => foundries.push(foundry.id()),
-                        Output::Basic(_) | Output::Treasury(_) => {}
-                    }
-                    let unlock_conditions = output_data
-                        .output
-                        .unlock_conditions()
-                        .expect("output must have unlock conditions");
-                    let sdr_amount = unlock_conditions
-                        .storage_deposit_return()
-                        .map(|sdr| sdr.amount())
-                        .unwrap_or(0);
-
-                    amount += output_data.output.amount() - sdr_amount;
-                }
+    for output_data in unspent_outputs.into_iter() {
+        // Panic: cannot fail for outputs belonging to an account.
+        let required_address = output_data
+            .output
+            .required_and_unlocked_address(current_time, &output_data.output_id, None)
+            .unwrap()
+            .0;
+
+        if address.inner() == &required_address {
+            outputs.push((output_data.output_id, output_data.output.kind_str().to_string()));
+
+            if let Some(nts) = output_data.output.native_tokens() {
+                native_tokens.add_native_tokens(nts.clone())?;
+            }
+
+            match &output_data.output {
+                Output::Alias(alias) => aliases.push(alias.alias_id_non_null(&output_data.output_id)),
+                Output::Nft(nft) => nfts.push(nft.nft_id_non_null(&output_data.output_id)),
+                Output::Basic(_) | Output::Foundry(_) | Output::Treasury(_) => {}
             }
+
+            let sdr_amount = output_data
+                .output
+                .unlock_conditions()
+                .unwrap()
+                .storage_deposit_return()
```

Comment:
what if storage reposit return gets returned to one of your addresses? :) should we handle that, or is it taken into account elsewhere?

Reply:
You can really do anything about it, it's not yours to unlock. But I think we'll completely change this function when we have a proper balance per address feature and that would somewhat be included in it